src/commonMain/kotlin/uuid.kt

```diff
@@ -158,6 +165,39 @@ public class Uuid(val uuid: ByteArray) {
             return Uuid(bytes)
         }
 
+        /**
+         * Parses a UUID from a String
+         *
+         * @param from The String, from which to deserialize the UUID
+         * @return a UUID, if the string is a valid UUID string or throws an [IllegalArgumentException]
+         */
+        fun fromString(from: String): Uuid {
+            if (from.length != UUID_STRING_LENGTH) {
+                throw IllegalArgumentException("Uuid string has invalid length: $from")
+            }
+            if (hyphenIndices.find { from[it] != '-' } != null) {
+                throw IllegalArgumentException("Uuid string has invalid format: $from (")
+            }
```

Comment:
nit (+ improved readability with `.all`):
```kotlin
require(hyphenIndices.all { from[it] == '-' }) {
    "Uuid string has invalid format: $from") 
}
```

Reply:
This looks much better. Done!